src/devices/niko.ts

```diff
@@ -23,66 +58,78 @@ const local = {
                     //       a hold/release cycle = three separate values, 16, 32, and 48
-                    const actionMap: KeyValue =
+
+                    // https://github.com/Koenkk/zigbee2mqtt/issues/13737#issuecomment-1520002786
+                    const buttonShift: {[key: string]: number} =
                         model.model == '552-721X1'
                             ? {
-                                  16: null,
-                                  64: 'single',
-                                  32: 'hold',
-                                  48: 'release',
-                                  256: null,
-                                  1024: 'single_ext',
-                                  512: 'hold_ext',
-                                  768: 'release_ext',
+                                  left: 4,
+                                  left_ext: 8,
+                                  right: 12,
+                                  right_ext: 16,
                               }
                             : {
-                                  16: null,
-                                  64: 'single_left',
-                                  32: 'hold_left',
-                                  48: 'release_left',
-                                  256: null,
-                                  1024: 'single_left_ext',
-                                  512: 'hold_left_ext',
-                                  768: 'release_left_ext',
-                                  4096: null,
-                                  16384: 'single_right',
-                                  8192: 'hold_right',
-                                  12288: 'release_right',
-                                  65536: null,
-                                  262144: 'single_right_ext',
-                                  131072: 'hold_right_ext',
-                                  196608: 'release_right_ext',
+                                  '': 4,
+                                  ext: 8,
                               };
+                    const actions: {[key: string]: number} = {
+                        null: 1,
+                        single: 4,
+                        hold: 2,
+                        release: 3,
+                    };
 
-                    state['action'] = actionMap[msg.data.switchAction];
+                    for (const button in buttonShift) {
+                        const shiftedValue = (msg.data.switchAction >> buttonShift[button]) & 0xf;
+                        for (const action in actions) {
+                            if (shiftedValue == actions[action]) {
+                                const buttonPostFix = button === '' ? '' : '_' + button;
+                                if ('null' === action) {
+                                    // publish without button name
+                                    publish({action: 'null'});
```

Comment:
I wonder if we should actually push the null values or if we should just skip them. I added them to be backwards compatible.

Reply:
I think we should not

Reply:
I have removed it